types/react-router/index.d.ts

```diff
@@ -61,7 +61,7 @@ declare module 'react-router' {
 
   interface RouteProps {
     location?: H.Location;
-    component?: React.SFC<RouteComponentProps<any> | void | {}> | React.ComponentClass<RouteComponentProps<any> | void | {}>;
```

Comment:
I was just pointing out that `{}` allows all non-null values, so allowing anything else in the union is redundant. This probably means that `{}` is the wrong type to use here. See @huy-nguyen's comment.